### PR TITLE
Java development

### PR DESCRIPTION
**Pull request by bx-h**
### Motivation and Context
Organize some key concepts in example code

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by johnoliver** [comment]
@bx-h thank you very much for the contribution looks good overall, I will perform a full review on it, probably early next week. Will probably be a few small changes

File: java/samples/semantickernel-concepts/README.md [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+## How to Define a Kernel?
+
+A `Kernel` is created using the `KernelBuilder`, where various services and plugins are configured via `withXXX()`.
+
+Create a Kernel using `KernelBuilder` and configure the necessary parameters
+
+```java
+Kernel kernel = Kernel.builder()
+    .withPlugin()
+    .withAIService()
```
Comment (johnoliver) [inline java/samples/semantickernel-concepts/README.md:61]: would change this to
```
    .withPlugin(myPlugin)
    .withAIService(openAiChatService)
```

File: java/samples/semantickernel-concepts/README.md [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+        .withDefaultExecutionSettings(
+                PromptExecutionSettings.builder()
+                        .withMaxTokens(100)
+                        .build())
+        .withTemplateFormat(PromptTemplateConfig.SEMANTIC_KERNEL_TEMPLATE_FORMAT)
+        .withName("translator")
+        .build();
+```
+
+> `KernelFunction.createFromPrompt(message)` is entirely equivalent to `KernelFunctionFromPrompt.builder().withTemplate(promptTemplate)` according to the source code.
```
Comment (johnoliver) [inline java/samples/semantickernel-concepts/README.md:159]: The formatting here seems off, is this intentional?

File: java/samples/semantickernel-concepts/README.md [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+
+> `SEMANTIC_KERNEL_TEMPLATE_FORMAT` refers to '**semantic-kernel**', which is a rendering engine for `{{$variable}}`
+> 
+> Another rendering engine is **'handlebars'**, which is used for `{{variable}}`
```
Comment (johnoliver) [inline java/samples/semantickernel-concepts/README.md:163]: Can link to:

https://devblogs.microsoft.com/semantic-kernel/using-handlebars-planner-in-semantic-kernel/

and

https://learn.microsoft.com/en-us/semantic-kernel/prompts/prompt-template-syntax

**Comment by johnoliver** [comment]
@bx-h left a few small comments, but overall looks great, thanks

**Comment by bx-h** [comment]
@johnoliver All comments resolved. Thanks.

**Review by johnoliver — APPROVED** [review]

**Pull request merged**